src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -319,11 +308,9 @@ public function denormalize(mixed $data, string $type, string $format = null, ar
         $normalizedData = $this->prepareForDenormalization($data);
         $extraAttributes = [];
 
-        $reflectionClass = new \ReflectionClass($type);
-        $object = $this->instantiateObject($normalizedData, $type, $context, $reflectionClass, $allowedAttributes, $format);
-        $resolvedClass = $this->objectClassResolver ? ($this->objectClassResolver)($object) : $object::class;
+        $mappedClass = $this->getMappedClass($normalizedData, $type, $context);
```

Comment:
warning this breaks code where the doctrine discriminator has no property (ie exists only in the database)